packages/google_adsense/lib/src/js_interop/adsbygoogle.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:js_interop';
+
+/// Binding to the `adsbygoogle` JS global.
+///
+/// See: https://support.google.com/adsense/answer/9274516?hl=en&ref_topic=28893&sjid=11495822575537499409-EU
+@JS('adsbygoogle')
+external AdsByGoogle get adsbygoogle;
+
+/// The Dart definition of the `adsbygoogle` global.
+@JS()
+@staticInterop
+abstract class AdsByGoogle {}
+
+/// The `adsbygoogle` methods mappings
+extension AdsByGoogleExtension on AdsByGoogle {
+  /// Replacement of part of the adUnit code:
+  /// <script>
+  // (adsbygoogle = window.adsbygoogle || []).push({});
+  // </script>
+  external void push(JSObject? params);
```

Comment:
In this case, since `adsbygoogle` is a "JSArray", if your JS interop type is like:

```
@JS('adsbygoogle')
extension type AdsByGoogle._(JSArray _) implements JSArray {
```

You may get the "push" method for free (with the exact same definition that you have).

`push` can also be made "private", so Dart users only see the `requestAd()` method, like this:

```
  @JS('push')
  external void _push(JSObject? params);
```

(Then `requestAd()` would call `_push` instead of `push`, of course).

Reply:
@ditman Interestingly when using the new interop style Im getting the following error from time to time:
`Instance of 'LegacyJavaScriptObject': type 'LegacyJavaScriptObject' is not a subtype of type JSArray<JSObject?>`
Im not quite sure what is causing it and it is not happening all the time